wandb/apis/public.py

```diff
@@ -2770,20 +2771,26 @@ def size(self):
         check_retry_fn=util.no_retry_auth,
         retryable_exceptions=(RetryError, requests.RequestException),
     )
-    def download(self, root=".", replace=False):
+    def download(self, root: str = ".", replace: bool = False, exist_ok: bool = False) -> io.TextIOWrapper:
         """Downloads a file previously saved by a run from the wandb server.
 
         Arguments:
             replace (boolean): If `True`, download will overwrite a local file
                 if it exists. Defaults to `False`.
             root (str): Local directory to save the file.  Defaults to ".".
+            exist_ok (boolean): If `True`, will not raise ValueError if file already
+                exists and will not re-download unless replace=True. Defaults to `False`.
 
         Raises:
-            `ValueError` if file already exists and replace=False
+            `ValueError` if file already exists and replace=False and exist_ok=False.
         """
         path = os.path.join(root, self.name)
-        if os.path.exists(path) and not replace:
-            raise ValueError("File already exists, pass replace=True to overwrite")
+        if os.path.exists(path):
+            if not replace and not exist_ok:
+                raise ValueError("File already exists, pass replace=True to overwrite")
+            elif not replace and exist_ok:
+                return open(path)
```

Comment:
total nit: (up to you)
```suggestion
        if os.path.exists(path) and not replace:
            if not exist_ok:
                raise ValueError("File already exists, pass replace=True to overwrite")
            else:
                return open(path)
```